### DISCUSSION TITLE
Modelling Namibia

### DISCUSSION THREAD

**Discussion by Constable1819** [pypsa-meets-earth/pypsa-earth#1110] [General]
Hi, 

I am having trouble running my custom country, Namibia. I got to run the Nigeria-Benin tutorial, followed the steps provided by https://pypsa-earth.readthedocs.io/en/latest/customization_basic1.html using the config.default.yaml file. I mainly changed the Countries to 'NA' and to a shorter period like the Nigeria-Benin example. what else has to be changed?

**Comment by ekatef** [comment]
Hey @Constable1819, thanks for reporting your progress!
Just to be sure, have you still go trough [the tutorial](https://pypsa-earth.readthedocs.io/en/latest/short_tutorial.html) itself? That is a recommended step before customising the model.

**Reply by Constable1819** [reply]
Yes I have.

**Reply by ekatef** [reply]
> Yes I have.

Perfect! Then, changing the country code should be enough. All other parameters can stay default, only `retrieve_databundle` should be switched-on for the first run, as described in the tutorial.

Additionally, it could make sense to specify a `name` under `run` ([this fragment](https://github.com/pypsa-meets-earth/pypsa-earth/blob/7b1249a20e6b56815329cc05ebf781e90b99d01f/config.default.yaml#L38-L39) of the configuration file) to keep the results of the model runs organised. That is not obligatory, just more convenient

**Reply by Constable1819** [reply]
Thank you

**Reply by Constable1819** [reply]
**This is the new error I am getting**

ValueError: array length 1 does not match index length 0
Error in rule build_osm_network:
    jobid: 8
    input: resources/osm/clean/all_clean_generators.geojson, resources/osm/clean/all_clean_lines.geojson, resources/osm/clean/all_clean_substations.geojson, resources/shapes/country_shapes.geojson
    output: resources/base_network/all_lines_build_network.csv, resources/base_network/all_converters_build_network.csv, resources/base_network/all_transformers_build_network.csv, resources/base_network/all_buses_build_network.csv
    log: logs/build_osm_network.log (check log file(s) for error details)

RuleException:
CalledProcessError in file /Users/xxx/pypsa-earth/Snakefile, line 236:
Command 'set -euo pipefail;  /opt/miniconda3/envs/pypsa-earth/bin/python3.10 /Users/xxx/pypsa-earth/.snakemake/scripts/tmpungexj6c.build_osm_network.py' returned non-zero exit status 1.
  File "/Users/xxx/pypsa-earth/Snakefile", line 236, in __rule_build_osm_network
  File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-09-27T001449.806531.snakemake.log

**Reply by ekatef** [reply]
> **This is the new error I am getting**
> 
> ValueError: array length 1 does not match index length 0 Error in rule build_osm_network: jobid: 8 input: resources/osm/clean/all_clean_generators.geojson, resources/osm/clean/all_clean_lines.geojson, resources/osm/clean/all_clean_substations.geojson, resources/shapes/country_shapes.geojson output: resources/base_network/all_lines_build_network.csv, resources/base_network/all_converters_build_network.csv, resources/base_network/all_transformers_build_network.csv, resources/base_network/all_buses_build_network.csv log: logs/build_osm_network.log (check log file(s) for error details)
> 
> RuleException: CalledProcessError in file /Users/xxx/pypsa-earth/Snakefile, line 236: Command 'set -euo pipefail; /opt/miniconda3/envs/pypsa-earth/bin/python3.10 /Users/xxx/pypsa-earth/.snakemake/scripts/tmpungexj6c.build_osm_network.py' returned non-zero exit status 1. File "/Users/xxx/pypsa-earth/Snakefile", line 236, in __rule_build_osm_network File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/concurrent/futures/thread.py", line 58, in run Shutting down, this might take some time. Exiting because a job execution failed. Look above for error message Complete log: .snakemake/log/2024-09-27T001449.806531.snakemake.log

Thanks a lot for reporting it. Just to be sure: have you removed all the previous output from the tutorial run? There is a more detailed explanation on that at the end of [short tutorial](https://pypsa-earth.readthedocs.io/en/latest/short_tutorial.html) section.

In case it won't help, could you please share the all error traceback you get from Snakemake run?

**Comment by kbuni123** [comment]
@Constable1819 @ekatef I am also getting the same issue when running the model for Ethiopia and Kenya. Could you please update me how the issue was resolved?

**Reply by ekatef** [reply]
> @Constable1819 @ekatef I am also getting the same issue when running the model for Ethiopia and Kenya. Could you please update me how the issue was resolved?

Hello and thanks a lot for letting know!
Could you please share all the traceback you get for the error you face, and a `config.yaml` you are using?

**Reply by kbuni123** [reply]
for config.yaml file I coppied the config.default.yaml. The only thing I changed is the name of the countries ("ET" ,"KE").

**Reply by kbuni123** [reply]
I could not upload my config.yaml. It says it is not supported when I try to upload it. However, I have attached the snake make and build_osm_work logs below.

[2025-01-19T194032.043147.snakemake.log](https://github.com/user-attachments/files/18474603/2025-01-19T194032.043147.snakemake.log)
[build_osm_network.log](https://github.com/user-attachments/files/18474611/build_osm_network.log)

**Reply by ekatef** [reply]
Hey @kbuni123, thank you for sharing the error traceback. To share the config, you can save it as txt file.

The error has not been seen before, and it's likely that it's reason is some local settings of the model. Some debug may be recommended to encounter the particular reason.

A couple of questions which can help to find a quick fix. Is the tutorial working for your setup? Have you tried removed all the output of the previous runs?
